### PR TITLE
Use `VmrDependencyUpdate` everywhere

### PR DESCRIPTION
**Pull request by premun**
This PR simplifies how we pass around related metadata that we need during VMR sync.
No functional changes but preparing the ground for a follow-up PR.

https://github.com/dotnet/arcade/issues/11912

### CODE REVIEW

**Review by oleksandr-didyk — APPROVED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyTracker.cs [hunk 1]
```diff
@@ -56,32 +56,32 @@ public class VmrDependencyTracker : IVmrDependencyTracker
...
 
-    public void UpdateDependencyVersion(SourceMapping mapping, VmrDependencyVersion version)
+    public void UpdateDependencyVersion(VmrDependencyUpdate update)
     {
         // TODO: https://github.com/dotnet/source-build/issues/2250
-        if (version.PackageVersion is null)
+        if (update.TargetVersion is null)
         {
-            version = version with { PackageVersion = DefaultVersion };
+            update = update with { TargetVersion = DefaultVersion };
```
Comment (andriipatsula) [inline src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyTracker.cs:64]: Nit: `DefaultVersion` is only used here. We can make it only visible inside `UpdateDependencyVersion` method.

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs [hunk 2]
```diff
@@ -113,21 +113,17 @@ public class VmrInitializer : VmrManagerBase, IVmrInitializer
...
 
-    private async Task InitializeRepository(
-        SourceMapping mapping,
-        string? targetRevision,
-        string? targetVersion,
-        CancellationToken cancellationToken)
+    private async Task InitializeRepository(VmrDependencyUpdate update, CancellationToken cancellationToken)
     {
-        _logger.LogInformation("Initializing {name} at {revision}..", mapping.Name, targetRevision ?? mapping.DefaultRef);
+        _logger.LogInformation("Initializing {name} at {revision}..", update.Mapping.Name, update.TargetRevision);
```
Comment (andriipatsula) [inline src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs:118]: I think, it's possible to have:
```C#
_logger.LogInformation($"Initializing '{update.Mapping.Name}' at '{update.TargetRevision}'..");
``

Anyway,  please put `name` and `revision` inside `` quotes.

Comment (andriipatsula) [reply src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs:118]: the same for Line: 161

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyUpdate.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public record VmrDependencyUpdate(
```
Comment (andriipatsula) [inline src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyUpdate.cs:10]: 'd be nice to have a short comment to a public record

**Review by andriipatsula — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs (hunk 2 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrInitializer.cs:118]: Ideally, you want to keep the message as a static string with placeholders and keep these as separate parameters (Visual Studio even tells you this) for few reasons, main ones being:
- No interpolation happens unnecessarily when logging doesn't happen (e.g. log level < information)
- The logger could do something with those values (e.g. store them in a separate column so they are searchable)

I used to have these quoted but for this specific statement, I keep copying the SHA a lot from the log and it was better without the quotes actually, so while I usually quote values in traces, here it's not too user friendly.

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyTracker.cs (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyTracker.cs:64]: I'd rather keep it up and more visible as it's quite important

File: src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyUpdate.cs (hunk 3 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrDependencyUpdate.cs:18]: Done

**Review by MilenaHristova — APPROVED** [review]

**Review by MilenaHristova — APPROVED** [review]

**Pull request merged**